cli/tests/pcluster/templates/test_dev_settings.py

```diff
@@ -50,3 +50,36 @@ def test_custom_cookbook(mocker, test_datadir, config_file_name):
 
     assert_that(effects).contains_only("Allow")
     assert_that(actions).contains_only("s3:GetObject", "s3:GetBucketLocation")
+
+
+@pytest.mark.parametrize(
+    "config_file_name",
+    [
+        "config.yaml",
+    ],
+)
+def test_custom_munge_key_iam_policy(mocker, test_datadir, config_file_name):
+    mock_aws_api(mocker)
+    mock_bucket_object_utils(mocker)
+    mocker.patch(
+        "pcluster.aws.secretsmanager.SecretsManagerClient.get_secret_value",
+        return_value={"SecretString": "validBase641234567890123456789012345678901234567"},
+    )
```

Comment:
Why is the mock for `get_secret_value` necessary to test the IAM policy? Doesn't it just use the Secret ARN? Have you tried commenting this mock out? Do we run the validators when generating the template?

Reply:
> Have you tried commenting this mock out?

Just tried. You are correct! Thanks for your reminding! The test can pass with only keeping these two line.
```
    mock_aws_api(mocker)
    mock_bucket_object_utils(mocker)
```
and delete the the mock
```
    mocker.patch(
        "pcluster.aws.secretsmanager.SecretsManagerClient.get_secret_value",
        return_value={"SecretString": "validBase641234567890123456789012345678901234567"},
    )
```

> Do we run the validators when generating the template?

No, I don't think so.